rr_mapper/src/laser_mapper.cpp

```diff
@@ -22,4 +24,447 @@ LaserMapper::LaserMapper(ros::NodeHandle nh) {
  * @return NONE
  */
 LaserMapper::~LaserMapper() {
+    belief_map_.clear();
 }
+
+/**
+ * @name PublishMap
+ * @brief Joins the entire map together
+ *        & Publishes the full_map
+ * @return NONE
+ */
+void LaserMapper::PublishMap() {
+    nav_msgs::OccupancyGrid full_map;
+
+    //Checks for lidar msg
+    int n = floor(abs(min_angle_-lidar_msg_.angle_min)/lidar_msg_.angle_increment);
+    double increment = lidar_msg_.angle_increment;
+
+    if (prev_header_.seq != lidar_msg_.header.seq)
+    {
+    for (double i = min_angle_; i < max_angle_; i+= increment)
+    {
+        // Check for NaN ranges
+        if (std::isnan (lidar_msg_.ranges[n]) == false)
+        {
+          DrawLine(i, lidar_msg_.ranges[n], inflate_obstacle_);
+        }
+        n++;
+    }
+    prev_header_ = lidar_msg_.header;
+    } 
+
+    // Initialize the new map to be published
+    full_map.header.frame_id = "/base_link";
+    full_map.info.resolution = map_res_;
+    full_map.info.width = map_width_;
+    full_map.info.height = map_height_;
+    full_map.info.origin.position.x = -map_width_*map_res_ / 2;//map_height_ * map_res_;
+    full_map.info.origin.position.y = 0;//-map_width_ * map_res_ / 2;
+    full_map.info.origin.orientation =
+                tf::createQuaternionMsgFromRollPitchYaw(0, 0, 0);
+    full_map.data.resize(map_height_*map_width_);
+
+    for (int i = 0; i < map_height_*map_width_; i++)
+    {
+        full_map.data[i] = belief_map_[i];
```

Comment:
I'm not entirely sure that this "belief_map" is, is it basically the previously stored lidar data, had some trouble following the logic to see how this was populated from the laser scan ros message that comes from our lidar

Reply:
uses updatelasermap function to place values at specific x,y coords